> Thanks @dbczumar ! but I see there is a test checking the exception I've removed
> 
> FAILED tests/pyfunc/test_mlserver.py::test_raise_error_mlserver_timeout[params0-Timeout is not yet supported in MLServer.]
> 
> So please let me first remove this test and then can be merged - would it be preferred to keep the test and skip it or just remove it?

I think removing the test is perfect! :)

--- a/mlflow/environment_variables.py
+++ b/mlflow/environment_variables.py
@@ -176,7 +176,7 @@ MLFLOW_REQUIREMENTS_INFERENCE_TIMEOUT = _EnvironmentVariable(
     "MLFLOW_REQUIREMENTS_INFERENCE_TIMEOUT", int, 120
 )
 
-#: Configuring the scoring server request timeout via environment variable
+#: Specifies the MLflow Model Scoring server request timeout in seconds
 #: (default: ``60``)
 MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT = _EnvironmentVariable(
     "MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT", int, 60

--- a/mlflow/pyfunc/__init__.py
+++ b/mlflow/pyfunc/__init__.py
@@ -225,6 +225,7 @@ import yaml
 
 import mlflow
 import mlflow.pyfunc.model
+from mlflow.environment_variables import MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT
 from mlflow.exceptions import MlflowException
 from mlflow.models import Model, ModelSignature, ModelInputExample
 from mlflow.models.model import MLMODEL_FILE_NAME
@@ -1076,7 +1077,7 @@ def spark_udf(spark, model_uri, result_type="double", env_manager="local"):
                 model_uri=local_model_path_on_executor,
                 port=server_port,
                 host="127.0.0.1",
-                timeout=60,
+                timeout=MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT.get(),
                 enable_mlserver=False,
                 synchronous=False,
                 stdout=subprocess.PIPE,

--- a/mlflow/pyfunc/backend.py
+++ b/mlflow/pyfunc/backend.py
@@ -6,6 +6,7 @@ import posixpath
 import sys
 import warnings
 
+from mlflow.environment_variables import MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT
 from mlflow.models import FlavorBackend
 from mlflow.models.docker_utils import (
     _build_image,
@@ -158,6 +159,7 @@ class PyFuncBackend(FlavorBackend):
         Serve pyfunc model locally.
         """
         local_path = _download_artifact_from_uri(model_uri)
+        timeout = timeout or MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT.get()
 
         server_implementation = mlserver if enable_mlserver else scoring_server
         command, command_env = server_implementation.get_cmd(

--- a/mlflow/pyfunc/mlserver.py
+++ b/mlflow/pyfunc/mlserver.py
@@ -1,6 +1,9 @@
+import logging
 import os
 
 from typing import Tuple, Dict
+
+_logger = logging.getLogger(__name__)
 
 MLServerMLflowRuntime = "mlserver_mlflow.MLflowRuntime"
 MLServerDefaultModelName = "mlflow-model"
@@ -20,7 +23,7 @@ def get_cmd(
         cmd_env["MLSERVER_HOST"] = host
 
     if timeout:
-        raise Exception("Timeout is not yet supported in MLServer.")
+        _logger.warning("Timeout is not yet supported in MLServer.")
 
     cmd_env["MLSERVER_MODEL_NAME"] = MLServerDefaultModelName
 

--- a/mlflow/pyfunc/scoring_server/__init__.py
+++ b/mlflow/pyfunc/scoring_server/__init__.py
@@ -46,7 +46,6 @@ except ImportError:
     from mlflow.pyfunc import load_pyfunc as load_model
 from mlflow.protos.databricks_pb2 import BAD_REQUEST
 from mlflow.server.handlers import catch_mlflow_exception
-from mlflow.environment_variables import MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT
 from io import StringIO
 
 _SERVER_MODEL_PATH = "__pyfunc_model_path__"
@@ -365,7 +364,6 @@ def get_cmd(
     model_uri: str, port: int = None, host: int = None, timeout: int = None, nworkers: int = None
 ) -> Tuple[str, Dict[str, str]]:
     local_uri = path_to_local_file_uri(model_uri)
-    timeout = timeout or MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT.get()
     # NB: Absolute windows paths do not work with mlflow apis, use file uri to ensure
     # platform compatibility.
     if os.name != "nt":

--- a/mlflow/utils/cli_args.py
+++ b/mlflow/utils/cli_args.py
@@ -156,7 +156,11 @@ PORT = click.option(
 )
 
 TIMEOUT = click.option(
-    "--timeout", "-t", help="Timeout in seconds to serve a request (default: 60)."
+    "--timeout",
+    "-t",
+    envvar="MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT",
+    default=60,
+    help="Timeout in seconds to serve a request (default: 60).",
 )
 
 # We use None to disambiguate manually selecting "4"

